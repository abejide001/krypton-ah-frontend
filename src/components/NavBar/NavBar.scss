@import '../../styles/partials/colors';
@import '../../styles/partials/styles';

.ui.menu.tinyBlackNavBar {
  min-height: 2rem;
  height: 2rem;
  background-color: $black;
  border-radius: 0;
  padding-right: 6rem;

  .item {
    padding: 0.5rem;
  }
}

.ui.pointing.dropdown > .menu:after {
  top: -0.25em;
  left: 80%;
  margin: 0 0 0 -0.25em;
}

.ui.menu .pointing.dropdown.item .menu {
  margin-top: 0.2em;
  width: 14em;
}

.left.menu.whiteBarMenu .item {
  padding-right: 0;
  padding-left: 1rem;
}

.ui.borderless.whiteNavBar.menu {
  margin: 0;
  padding: 0 7.4375em;
  border-radius: 0;
  height: 5rem;

  img.ui.image.ahIcon {
    width: 3.2em;
    font-size: 1rem;
  }

  .whiteBarMenu h1 {
    font-weight: bolder;
    font-size: 2.3em;
    margin-left: 0;
    padding-left: 0;
  }
  .left.menu.whiteBarMenu {
    .item {
      padding-left: 0.25em;
      padding-right: 0.25em;
    }
  }

  .right.menu {
    .item i {
      font-size: 1rem;
    }

    .item {
      .ui.avatar.image.avatarImg {
        order: 2;
        font-size: 1.5rem;
      }
      .dropdown.icon {
        order: 1;
        margin-right: 0.5rem;
      }
    }
  }
}
@media (max-width: 767px) {
  .ui.menu.tinyBlackNavBar {
    padding-right: 1rem;
  }

  .ui.borderless.whiteNavBar.menu {
    margin: 0;
    padding: 0 0.475em;

    img.ui.image.ahIcon {
      width: 3.2em;
      font-size: 1rem;
    }

    .whiteBarMenu h1 {
      font-weight: bolder;
      font-size: 2.3em;
      margin-left: 0;
      padding-left: 0;
      display: none;
    }
    .left.menu.whiteBarMenu {
      .item {
        padding-left: 0.25em;
        padding-right: 0.25em;
      }
    }

    .right.menu {
      .item i {
        font-size: 1rem;
      }

      .item {
        .ui.avatar.image.avatarImg {
          order: 2;
          font-size: 1.5rem;
        }
        .dropdown.icon {
          order: 1;
          margin-right: 0.5rem;
        }
      }
    }
  }
}

.Toastify__toast--success {
  background: #0d3d2c;
  padding-left: 2rem;
}
