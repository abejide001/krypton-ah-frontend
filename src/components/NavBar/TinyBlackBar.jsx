import React from 'react';
import PropTypes from 'prop-types';
import { Menu, Visibility } from 'semantic-ui-react';
import SocialMediaIcons from '../SocialMediaIcons';

export default function TinyBlackNavBar({ stickTopMenu, unStickTopMenu }) {
  return (
    <Visibility onBottomPassed={stickTopMenu} onBottomVisible={unStickTopMenu} once={false}>
      <Menu position="relative" className="tinyBlackNavBar">
        <SocialMediaIcons iconSize="small" position="right" />
      </Menu>
    </Visibility>
  );
}

TinyBlackNavBar.defaultProps = {
  stickTopMenu: null,
  unStickTopMenu: null
};

TinyBlackNavBar.propTypes = {
  stickTopMenu: PropTypes.func,
  unStickTopMenu: PropTypes.func
};
