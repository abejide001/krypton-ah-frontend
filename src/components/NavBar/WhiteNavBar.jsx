import React from 'react';
import PropTypes from 'prop-types';
import {
  Dropdown,
  Image,
  Menu,
  Icon,
  Button
} from 'semantic-ui-react';
import AHIcon from '../../images/Logo.png';
import avatarPlaceholder from '../../images/avatar.png';

export default function WhiteNavBar({ fixed, isAuthenticated, avatarUrl }) {
  return (
    <Menu fixed={fixed ? 'top' : null} borderless className="whiteNavBar">
      <Menu.Menu position="left" className="whiteBarMenu">
        <Menu.Item>
          <Image alt="AH-logo" className="ahIcon" src={AHIcon} />
        </Menu.Item>
        <Menu.Item>
          <h1>Authors Haven</h1>
        </Menu.Item>
      </Menu.Menu>
      <Menu.Menu position="right">
        <Menu.Item>
          <Icon fitted size="big" name="search" />
        </Menu.Item>
        <Menu.Item>
          <Button secondary>Write an Article</Button>
        </Menu.Item>
        {isAuthenticated ? (
          <Menu.Item>
            <Dropdown
              pointing
              simple
              item
              className="link item"
              trigger={<Image src={avatarUrl} avatar className="avatarImg" />}
            >
              <Dropdown.Menu className="right">
                <Dropdown.Item>PROFILE</Dropdown.Item>
                <Dropdown.Item>BOOKMARKS</Dropdown.Item>
                <Dropdown.Item>LOG OUT</Dropdown.Item>
              </Dropdown.Menu>
            </Dropdown>
          </Menu.Item>
        ) : null}
      </Menu.Menu>
    </Menu>
  );
}

WhiteNavBar.defaultProps = {
  avatarUrl: avatarPlaceholder,
  isAuthenticated: false,
  fixed: false
};

WhiteNavBar.propTypes = {
  fixed: PropTypes.bool,
  isAuthenticated: PropTypes.bool,
  avatarUrl: PropTypes.string
};
