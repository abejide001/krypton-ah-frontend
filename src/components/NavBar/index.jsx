import React from 'react';
import TinyBlackBar from './TinyBlackBar';
import { WhiteBar } from './WhiteNavBar';
import './NavBar.scss';

export default class NavBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = { fixed: false };
    this.stickTopMenu = this.stickTopMenu.bind(this);
    this.unStickTopMenu = this.unStickTopMenu.bind(this);
  }

  stickTopMenu() {
    this.setState({ fixed: true });
  }

  unStickTopMenu() {
    this.setState({ fixed: false });
  }

  render() {
    const { fixed } = this.state;
    return (
      <React.Fragment>
        <TinyBlackBar stickTopMenu={this.stickTopMenu} unStickTopMenu={this.unStickTopMenu} />
        <WhiteBar fixed={fixed} />
      </React.Fragment>
    );
  }
}
