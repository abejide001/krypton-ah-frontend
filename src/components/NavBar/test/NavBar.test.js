import React from 'react';
import { shallow } from 'enzyme';
import NavBar from '..';
import TinyBlackBar from '../TinyBlackBar';
import WhiteBar from '../WhiteNavBar';

describe('NavBar', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<NavBar />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should set state property fixed to true', () => {
    const wrapper = shallow(<NavBar />);
    wrapper.instance().unStickTopMenu();
    wrapper.instance().stickTopMenu();
    expect(wrapper.state().fixed).toEqual(true);
  });
  it('should set state property fixed to false', () => {
    const wrapper = shallow(<NavBar />);
    wrapper.instance().unStickTopMenu();
    expect(wrapper.state().fixed).toEqual(false);
  });
});

describe('TinyBlackBar', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<TinyBlackBar />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('WhiteNavBar', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<WhiteBar />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('WhiteNavBar with props', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<WhiteBar fixed isAuthenticated avatarUrl="https://fakeAvatar.com" />);
    expect(wrapper).toMatchSnapshot();
  });
});
