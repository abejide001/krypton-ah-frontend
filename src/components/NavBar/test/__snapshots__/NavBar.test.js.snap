// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match snapshot 1`] = `
<Fragment>
  <TinyBlackNavBar
    stickTopMenu={[Function]}
    unStickTopMenu={[Function]}
  />
  <WhiteNavBar
    avatarUrl="test-file-stub"
    fixed={false}
    isAuthenticated={false}
  />
</Fragment>
`;

exports[`TinyBlackBar should match snapshot 1`] = `
<Visibility
  context={[Window]}
  continuous={false}
  offset={
    Array [
      0,
      0,
    ]
  }
  onBottomPassed={null}
  onBottomVisible={null}
  once={false}
  updateOn="events"
>
  <Menu
    className="tinyBlackNavBar"
    position="relative"
  >
    <SocialMediaIcons
      iconSize="small"
      position="right"
    />
  </Menu>
</Visibility>
`;

exports[`WhiteNavBar should match snapshot 1`] = `
<Fragment>
  <ToastContainer
    autoClose={5000}
    bodyClassName={null}
    className={null}
    closeButton={
      <CloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="top-right"
    progressClassName={null}
    progressStyle={null}
    rtl={false}
    style={null}
    toastClassName={null}
    transition={[Function]}
  />
  <Menu
    borderless={true}
    className="whiteNavBar"
    fixed={null}
  >
    <MenuMenu
      className="whiteBarMenu"
      position="left"
    >
      <MenuItem>
        <Image
          alt="AH-logo"
          as="img"
          className="ahIcon"
          src="test-file-stub"
          ui={true}
        />
      </MenuItem>
      <MenuItem>
        <h1>
          Authors Haven
        </h1>
      </MenuItem>
    </MenuMenu>
    <MenuMenu
      position="right"
    >
      <MenuItem>
        <SearchModal />
      </MenuItem>
      <MenuItem>
        <Link
          replace={false}
          to="/login"
        >
          <Button
            as="button"
            secondary={true}
          >
            Write an Article
          </Button>
        </Link>
      </MenuItem>
    </MenuMenu>
  </Menu>
</Fragment>
`;

exports[`WhiteNavBar with props should match snapshot 1`] = `
<Fragment>
  <ToastContainer
    autoClose={5000}
    bodyClassName={null}
    className={null}
    closeButton={
      <CloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="top-right"
    progressClassName={null}
    progressStyle={null}
    rtl={false}
    style={null}
    toastClassName={null}
    transition={[Function]}
  />
  <Menu
    borderless={true}
    className="whiteNavBar"
    fixed="top"
  >
    <MenuMenu
      className="whiteBarMenu"
      position="left"
    >
      <MenuItem>
        <Image
          alt="AH-logo"
          as="img"
          className="ahIcon"
          src="test-file-stub"
          ui={true}
        />
      </MenuItem>
      <MenuItem>
        <h1>
          Authors Haven
        </h1>
      </MenuItem>
    </MenuMenu>
    <MenuMenu
      position="right"
    >
      <MenuItem>
        <SearchModal />
      </MenuItem>
      <MenuItem>
        <Link
          replace={false}
          to="/login"
        >
          <Button
            as="button"
            secondary={true}
          >
            Write an Article
          </Button>
        </Link>
      </MenuItem>
      <MenuItem>
        <Dropdown
          additionLabel="Add "
          additionPosition="top"
          className="link item"
          closeOnBlur={true}
          deburr={false}
          icon="dropdown"
          item={true}
          minCharacters={1}
          noResultsMessage="No results found."
          openOnFocus={true}
          pointing={true}
          renderLabel={[Function]}
          searchInput="text"
          selectOnBlur={true}
          selectOnNavigation={true}
          simple={true}
          trigger={
            <Image
              as="img"
              avatar={true}
              className="avatarImg"
              src="https://fakeAvatar.com"
              ui={true}
            />
          }
          wrapSelection={true}
        >
          <DropdownMenu
            className="right"
          >
            <DropdownItem>
              PROFILE
            </DropdownItem>
            <DropdownItem>
              BOOKMARKS
            </DropdownItem>
            <DropdownItem>
              LOG OUT
            </DropdownItem>
          </DropdownMenu>
        </Dropdown>
      </MenuItem>
    </MenuMenu>
  </Menu>
</Fragment>
`;
